cli/src/commands/duplicate.rs

```diff
@@ -164,6 +170,26 @@ pub(crate) fn cmd_duplicate(
             }
         }
     }
+
+    let mut new_descs = HashMap::new();
+
+    {
+        let template = tx
+            .settings()
+            .get_string("templates.duplicate_description")?;
+        let parsed = tx.parse_commit_template(ui, &template)?;
+
+        for commit_id in &to_duplicate {
+            let mut output = Vec::new();
+            let commit = tx.repo().store().get_commit(commit_id)?;
+            parsed
+                .format(&commit, &mut PlainTextFormatter::new(&mut output))
+                .expect("write() to vec backed formatter should never fail");
+
+            new_descs.insert(commit_id.clone(), output.into_string_lossy());
+        }
+    }
```

Comment:
nit: (at your option), now we can remove `mut`, but I don't have preference over these.
```rust
let new_descs: HashMap<_, _> = {
    ...
    to_duplicate.iter().map(|commit_id| -> BackendResult<_> { .. }).try_collect()?
};

Reply:
I'll think about it -- couldn't we do this before, though?

Reply:
> couldn't we do this before, though?

You could, but we would need something like `new_descs = if { ... } else { HashMap::new() };` to remove `mut`.